gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java

```diff
@@ -45,9 +46,12 @@ public String id() {
 
     @Override
     public void start() {
-        final String pem = new String(Base64.getDecoder().decode(subscription.getClientCertificate()));
-        final KeyStore keystore = KeyStoreUtils.initFromPemCertificate(pem, getPassword(), subscription.getId());
-        onEvent(new KeyStoreEvent.LoadEvent(id(), keystore, getPassword()));
+        if (!started) {
+            final String pem = new String(Base64.getDecoder().decode(subscription.getClientCertificate()));
+            final KeyStore keystore = KeyStoreUtils.initFromPemCertificate(pem, getPassword(), subscription.getId());
+            onEvent(new KeyStoreEvent.LoadEvent(id(), keystore, getPassword()));
+            started = true;
```

Comment:
You should probably put it at the top level to avoid a concurrent start() (or use AtomicBoolean so you are sure there is 1  single start)